src/posthog-core.js

```diff
@@ -722,7 +732,13 @@ PostHogLib.prototype._register_single = function (prop, value) {
  * @param {Object|String} options (optional) If {send_event: false}, we won't send an $feature_flag_call event to PostHog.
  */
 PostHogLib.prototype.isFeatureEnabled = function (key, options = {}) {
-    return this.feature_flags.isFeatureEnabled(key, options)
+    if (!this.featureFlags) {
```

Comment:
This change has a couple of issues:

1. Returning undefined makes no sense and can break peoples code. This function should return a boolean. `false` is the good default
2. Checking `!this.featureFlags` is misleading as it will fail if no flags are turned on.
3. This will cause logspam - this is always reached if isFeatureEnabled is called before decide returns a response.
4. This doesn't belong to posthog-core.js at all - feature flags file should handle all the logic here. Otherwise the two ways to call featureflags in the library diverge in the implementation.

I suggest not including this change at all.

Reply:
Removed this code, users will get an error if attempting to load feature flags with `/decide` endpoint disabled anyways.